meson.build

```diff
@@ -470,6 +470,24 @@ if elogind.found()
   conf.set('HAVE_LOGIND' , '1')
 endif
 
+if get_option('launchd').disabled()
+  use_launchd = false
+else
+  launchctl = find_program('launchctl', required: get_option('launchd'))
+  use_launchd = cc.has_header('launch.h') and launchctl.found()
+  if get_option('launchd').enabled() and not use_launchd
+    error('launchd support requested but not found')
+  endif
+endif
```

Comment:
I think you're missing a `disable_auto_if` here when running on !macos

Reply:
you mean like:

    -  launchctl = find_program('launchctl', required: get_option('launchd'))
    +  launchctl = find_program('launchctl', required: get_option('launchd').disable_auto_if(host_machine.system() != 'darwin'))

perhaps?

Reply:
Yup